### PR TITLE
(enhancement): mother hen

### PR DESCRIPTION
**Pull request by Jamesbarford**

### CODE REVIEW

**Comment by Jamesbarford** [comment]
If we don't want mother hen, happy to close PR

File: Moonshot/baddies/BaddieLaser/BaddieSpawner.gd [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	baddie_instance.position = global_position -  Vector2(0.75, 0.75) # slightly beneath mother hen
+	baddie_instance.scale = Vector2(0.75, 0.75) # smaller
+
+	_baddie_dictionary[baddie_instance.get_instance_id()] = baddie_instance
+
+	_add_baddie_to_scene(baddie_instance)
+	_clean_dictionary(baddie_dictionary_keys)
+
+func _is_max_capacity(baddie_dictionary_keys: Array) -> bool:
+	return baddie_dictionary_keys.size() > max_baddies
```
Comment (wattsit) [inline Moonshot/baddies/BaddieLaser/BaddieSpawner.gd:35]: Maybe when spawned baddies are killed they're removed from the dictionary so more can spawn. Not sure if this is an easy change however.

**Comment by wattsit** [comment]
Maybe we want mother hen to damage the player on contact? Currently after killing her brood she just a bit of a floating meat bag :P

**Comment by kepler471** [comment]
Love this enemy. I think she needs some slightly altered visuals, the babies need different visuals too and prob shouldn't collide with each other ? Also spawning in waves with maybe a different attack type and a vulnerable moment in between the waves would add a lot to how it feels to fight her.

File: Moonshot/baddies/BaddieLaser/BaddieSpawner.gd (hunk 1 above)
Comment (Jamesbarford) [reply Moonshot/baddies/BaddieLaser/BaddieSpawner.gd:35]: `_clean_dictionary ` does this:

image


 This is an implementation of a [ring buffer](https://en.wikipedia.org/wiki/Circular_buffer) to ensure the total number of bear boi's never exceeds `max_baddies `. 

And when a free space is available it gets filled

**Comment by whitew1994WW** [comment]
Just an idea, can we have 'mother hen' decrease in size (sprite) as its health reduces?

**Pull request merged**